### PR TITLE
graphql-voyager: bump dep to 2.0 + add dev setup

### PR DESCRIPTION
**Pull request by Rugvip**
## Hey, I just made a Pull Request!

This bump breaks part of the styling of the plugin, but we can't keep it on the 1.0 RC as it brings in some old dependencies like MUI 3. Will leave it to additional contributions to fix the style issues. The plugin is still somewhat functional.

CC @MitchWijt in case you want to have a look

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by MitchWijt** [comment]
Awesome, looks good. I guess we can close my PR then right?

**Comment by Rugvip** [comment]
Also CC @adamdmharvey, because #19255 😅

I'll spend a bit of time digging into the CSS type issues, but don't wanna get too distracted x)

File: yarn.lock [hunk 1]
```diff
@@ -22583,17 +22516,17 @@ __metadata:
...
   linkType: hard
 
-"csstype@npm:^3.0.2, csstype@npm:^3.0.6, csstype@npm:^3.1.2":
-  version: 3.0.9
-  resolution: "csstype@npm:3.0.9"
-  checksum: 199f9af7e673f9f188525c3102a329d637ff46c52f6385a4427ff5cb17adcb736189150170a7af7c5701d18d7704bdad130273f4aa7e44c6c4f9967e6115dc93
+"csstype@npm:^3.0.2, csstype@npm:^3.0.6, csstype@npm:^3.1.1, csstype@npm:^3.1.2":
+  version: 3.1.2
+  resolution: "csstype@npm:3.1.2"
+  checksum: e1a52e6c25c1314d6beef5168da704ab29c5186b877c07d822bd0806717d9a265e8493a2e35ca7e68d0f5d472d43fac1cdce70fd79fd0853dff81f3028d857b5
```
Comment (tudi2d) [inline yarn.lock:22529]: Heads-up that I ran into a similar [problem](https://github.com/backstage/backstage/pull/15484#discussion_r1179025947) here a while back, where `csstype` brought in a breaking change in a `patch`.

File: packages/theme/src/v4/types.ts [hunk 2]
```diff
@@ -104,3 +105,21 @@ declare module '@material-ui/core/styles/createTheme' {
...
+//
+// This makes sure that the font weights in the theme are compatible with the styles.
+declare module '@material-ui/core/styles/createTypography' {
+  interface Typography {
+    fontWeightLight: CSSProperties['fontWeight'];
+    fontWeightRegular: CSSProperties['fontWeight'];
+    fontWeightMedium: CSSProperties['fontWeight'];
+    fontWeightBold: CSSProperties['fontWeight'];
+  }
+}
```
Comment (tudi2d) [inline packages/theme/src/v4/types.ts:125]: interesting. I'm not sure it's v4-specific, but rather the combination of v4 & v5 I would assume. Would be happy with this work around, if it fixes `csstypes` issue for now!

**Review by tudi2d — COMMENTED** [review]
good idea - let's see how this can be fixed

File: plugins/graphql-voyager/package.json [hunk 3]
```diff
@@ -35,7 +35,7 @@
     "@material-ui/core": "^4.12.2",
     "@material-ui/icons": "^4.9.1",
     "@material-ui/lab": "4.0.0-alpha.61",
-    "graphql-voyager": "^1.0.0-rc.31",
+    "graphql-voyager": "^2.0.0",
```
Comment (pjungermann) [inline plugins/graphql-voyager/package.json:38]: FYI:
graphql-voyager v2.0.0 upgraded from _(previously at RC31)_ material-ui v4 to v5

```
    "@mui/icons-material": "5.11.0",
    "@mui/lab": "5.0.0-alpha.114",
    "@mui/material": "5.11.2",
```

PS: This PR could resolve #19327 and #19326.

The RC31 was dependent on `"@material-ui/core": ^3.9.3` -> `recompose: 0.28.0 - 0.30.0` -> `fbjs: ^0.8.1` -> `isomorphic-fetch: ^2.1.1` -> `node-fetch: ^1.0.1`

**Review by tudi2d — APPROVED** [review]
approving to get this in - looks good to me.

**Pull request closed without merging**